File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.java
class Solution {
    public int solution(int[] players, int m, int k) {
        int[] added = new int[k];
        int serverCount = 0;
        int expendCount = 0;
        
        for (int i = 0; i < players.length; i++) {
            serverCount -= added[i % k];
            added[i % k] = 0;

            int needed = players[i] / m;
            int toAdd = needed - serverCount;
            
            if (toAdd > 0) {
                added[i % k] = toAdd;
                serverCount += toAdd;
                expendCount += toAdd;
            }
        }
        
        return expendCount;
    }
}

File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.kt
class Solution {
    fun solution(players: IntArray, m: Int, k: Int): Int {
        val added = IntArray(k)
        var serverCount = 0
        var expendCount = 0
        
        for (i in players.indices) {
            serverCount -= added[i % k]
            added[i % k] = 0
            
            val needed = players[i] / m
            val toAdd = needed - serverCount
            
            if (toAdd > 0) {
                added[i % k] = toAdd
                serverCount += toAdd
                expendCount += toAdd
            }
        }
        
        return expendCount
    }
}